### ISSUE TITLE
Feature request : allow variable player number on a game/env

### ISSUE THREAD

**Issue by zorgluf** [davidADSP/SIMPLE#6]
Hi !
It would be nice to be able to choose the number of players for a game, although I don't master the impact on the training loop (one model per number of players ?).
Is there any other limitation to implement this feature ?
Thanks for your answer and for all the work done.

**Comment by davidADSP** [comment]
Yes agreed - it's tricky as usually the model input shape is dependent on the number of players (e.g. 4 player sushi go requires the agent to keep track of 4 different hands, so a larger input shape than for the 3 player game). Therefore, it's currently not possible to utilise a model trained for a 3 player game for a 4 player game, and therefore they may as well be completely separate games in the folder.

I'm open to ideas as to how a n_player agnostic agent would work - totally agree it would be a cool feature!

**Comment by aamonkeyA** [comment]
Wouldn't it be sufficient to always assume the max player count? Then you have the shape for 4-player sushi go and if you actually play with only 3 players, you zero out the extra players.
